tools/Dockerfile.dev

```diff
@@ -1,68 +1,36 @@
 ARG ROS_DISTRO=humble
 ARG PREFIX=
-ARG ROSBOT_FW_RELEASE=0.8.0
-## ============================ STM32FLASH =================================
-# stm32flash needs an older version of glibc (2.28), which is why ubuntu 18.04 was used
-FROM ubuntu:18.04 AS stm32flash_builder
-
-ARG ROS_DISTRO
-ARG ROSBOT_FW_RELEASE
-ARG TARGETARCH
-
-SHELL ["/bin/bash", "-c"]
-
-# official releases are only for intel archs, so we need to build stm32flash from sources
-RUN apt-get update && apt-get install -y \
-        curl
-
-RUN echo ros_distro=$ROS_DISTRO firmware_release=$ROSBOT_FW_RELEASE
-
-# Copy firmware binaries
-RUN curl -L https://github.com/husarion/stm32flash/releases/download/2023-12-05/stm32flash-$TARGETARCH -o /stm32flash && \
-    chmod +x /stm32flash
+ARG HUSARION_ROS_BUILD_TYPE=hardware
 
 ## =========================== ROS builder ===============================
 FROM husarnet/ros:${PREFIX}${ROS_DISTRO}-ros-base AS ros_builder
 
 ARG ROS_DISTRO
 ARG PREFIX
+ARG HUSARION_ROS_BUILD_TYPE
 
-SHELL ["/bin/bash", "-c"]
+ENV HUSARION_ROS_BUILD_TYPE=${HUSARION_ROS_BUILD_TYPE}
 
 WORKDIR /ros2_ws
 RUN mkdir src
 
-COPY rosbot src/rosbot
-COPY rosbot_bringup src/rosbot_bringup
-COPY rosbot_controller src/rosbot_controller
-COPY rosbot_description src/rosbot_description
-COPY rosbot_utils src/rosbot_utils
+COPY ./ src/
 
 RUN apt-get update && apt-get install -y \
         python3-pip \
         stm32flash
 
 RUN vcs import src < src/rosbot/rosbot_hardware.repos && \
-    cp -r src/ros2_controllers/diff_drive_controller src/ && \
-    cp -r src/ros2_controllers/imu_sensor_broadcaster src/ && \
-    rm -rf src/ros2_controllers && \
-    # without this line (using vulcanexus base image) rosdep init throws error: "ERROR: default sources list file already exists:"
+    if [ "$HUSARION_ROS_BUILD_TYPE" == "simulation" ]; then \
+        vcs import src < src/rosbot/rosbot_simulation.repos; \
+    else \
+        rm -rf src/rosbot_gazebo; \
```

Comment:
We can add conditions inside rosbot_gazebo like this:
```xml
  <exec_depend condition="($HUSARION_ROS_BUILD == simulation)">husarion_office_gz</exec_depend>
  <exec_depend condition="($HUSARION_ROS_BUILD == simulation)">ros_gz_sim</exec_depend>
  <!-- Ignition dependency is specified in the ros_gz_sim package,
    version can chosen using GZ_VERSION (or IGNITION_VERSION) env variable,
    for details refer to the ros_gz_sim package -->
  <exec_depend condition="($HUSARION_ROS_BUILD == simulation)">ros_gz_bridge</exec_depend>
  <exec_depend condition="($HUSARION_ROS_BUILD == simulation)">ign_ros2_control</exec_depend>
  <exec_depend condition="($HUSARION_ROS_BUILD == simulation)">nav2_common</exec_depend>
```

But maybe in the another PR.